server/src/main/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -144,6 +143,24 @@ public static long sizeOfRelocatingShards(
             }
         }
 
+        // Count the STARTED shards which are unaccounted in the cluster info for searchable snapshots
+        for (ShardRouting shard : node.shardsWithState(ShardRoutingState.STARTED)) {
+            IndexMetadata indexMetadata = metadata.index(shard.index());
+            if (indexMetadata == null) {
+                continue;
+            }
+            if (indexMetadata.isSearchableSnapshot() == false) {
+                continue;
+            }
+            if (reservedSpace.containsShardId(shard.shardId())) {
+                continue;
+            }
+            if (clusterInfo.getShardSize(shard) != null) {
+                continue;
+            }
+            totalSize += Math.max(shard.getExpectedShardSize(), 0L);
```

Comment:
I think we need this in the INITIALIZING case too for searchable snapshots? (or rather use the `shard.getExpectedShardSize` value in place of the default `0L` there).

I wonder if we can make a test that fail in that case. I.e., we have X initializing shards, allowing us to go X-1 shards too high.

Reply:
Yes, since we only account for relocating shards for the `INITIALIZING` shards. I agree that it would be nice to add a test for that case.

Reply:
I've changed `0L` to `Math.max(routing.getExpectedShardSize(), 0L)` for the `INITIALIZING` case, too, but I couldn't come up with a test case showing that it makes any difference :(